dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdExtension.kt

```diff
@@ -19,15 +19,27 @@ open class DdExtension : DdExtensionConfiguration() {
      */
     var enabled: Boolean = true
 
+    // introduced because cannot use lateinit with `variants` because of name
+    // ambiguity (method vs property). Consider switching to constructor injection
+    // via https://docs.gradle.org/current/javadoc/org/gradle/api/model/ObjectFactory.html#domainObjectContainer-java.lang.Class-
+    // once it is stable
+    private lateinit var _variants: NamedDomainObjectContainer<DdExtensionConfiguration>
+
     /**
      * Container for the variant's configurations.
      */
-    var variants: NamedDomainObjectContainer<DdExtensionConfiguration>? = null
+    var variants: NamedDomainObjectContainer<DdExtensionConfiguration>
+        get() = _variants
+        set(value) {
+            _variants = value
+        }
```

Comment:
Do you really need this property here? You're just using the existing `_variant` property without any additional logic.

Reply:
yes, because I cannot reference just `::variants.isInitialized`, because it creates ambiguity between `variants` property and `variants` method and compiler cannot resolve it.

Something like this doesn't work either:

```
val foo: KProperty<*> = ::variants
if (foo.isInitialized) { }
```

because Kotlin compiler allows `.isInitialized` call only directly. So either need to do such hack to remove ambiguity, or use `ObjectFactory` (which is incubating). Well, there is also an option to use different names for property and method, but not sure if it is the path we would like to take for the plugin API.